File: academic-system-java/src/main/java/org/example/model/Usuario.java
package org.example.model;

import lombok.Getter;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;

@Getter
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Usuario {
    @EqualsAndHashCode.Include
    private String username;
    private String password;
    private String role;
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Models/Usuario.cs
namespace SistemaAcademico.Models;

// Equivalente à classe Java: Usuario
// Lombok @Getter → propriedades C# com get
// Lombok @AllArgsConstructor → construtor gerado com todos os campos
// Lombok @EqualsAndHashCode(onlyExplicitlyIncluded = true) com @EqualsAndHashCode.Include em username
//   → Equals e GetHashCode baseados apenas em Username
public class Usuario
{
    public string Username { get; }
    public string Password { get; }
    public string Role { get; }

    public Usuario(string username, string password, string role)
    {
        Username = username;
        Password = password;
        Role = role;
    }

    // Equivalente a Lombok @EqualsAndHashCode(onlyExplicitlyIncluded = true) com Username incluso
    public override bool Equals(object? obj)
    {
        if (obj is Usuario other)
        {
            return string.Equals(Username, other.Username, StringComparison.Ordinal);
        }
        return false;
    }

    public override int GetHashCode()
    {
        return Username?.GetHashCode() ?? 0;
    }
}
